tf_sig_build_dockerfiles/setup.python.sh

```diff
@@ -33,9 +34,18 @@ ln -sf /usr/bin/$VERSION /usr/bin/python
 ln -sf /usr/lib/$VERSION /usr/lib/tf_python
 
 # Install pip
-curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-python3 get-pip.py
+if [[ "$VERSION" == "python3.10" ]]; then
+  # Python 3.10 pip reference is broken for pip 21.3 (pypa/pip#10647)
+  # TODO(rameshsampath): Remove once Python 3.10 works with latest pip
+  python3 -m ensurepip
```

Comment:
What happens if you replace this entire install-pip section with just this?

```
python3 -m ensurepip
python3 -m pip install --no-cache-dir --upgrade pip
```

I ran a quick test and didn't see any obvious problems. Can you give more details about the issue that's being avoided?

Reply:
In one of the earlier builds, where I had the kept the setup that was there in the docker I ran into one of two problems:
1. No module named pip (https://github.com/tensorflow/build/runs/4420931846?check_suite_focus=true#step:5:1893)
2. It installs the packages, but not in the right place and hence not able to find it when trying to use it.

Reply:
Also, Updating Ensurepip doesn't work for Python 3.8 since we are updating system python and gives an error message: https://github.com/tensorflow/build/runs/4463376066?check_suite_focus=true#step:5:1204